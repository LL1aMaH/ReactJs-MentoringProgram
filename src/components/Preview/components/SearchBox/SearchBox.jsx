import React, { memo, useState } from 'react';
// import { useHistory } from 'react-router-dom';

import { Button } from '../../../Button/Button';

import classnames from 'classnames';

import styles from '../../../../../styles/SearchBox.module.scss'


export const SearchBox = memo(function SearchBox(props) {
  const { searchValue: value = '', className } = props;
  const [searchValue, setSearchValue] = useState(value);

  // const history = useHistory();

  const handleChange = (e) =>
    setSearchValue(e.currentTarget.value);

  const handleClick = () => {
    // if (searchValue) history.push(`/search/${searchValue}`);
  };

  const searchBoxClassName = classnames(styles.searchBox, className);
  return (
    <div className={searchBoxClassName}>
      <input placeholder="What do you want to watch?" value={searchValue} onChange={handleChange} />
      <Button className={styles.button} onClick={handleClick} testId="button">
        SEARCH
      </Button>
    </div>
  );
});
