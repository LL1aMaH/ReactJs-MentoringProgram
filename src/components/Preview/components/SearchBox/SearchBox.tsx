import React, { memo } from 'react';

import { Button } from 'Components/Button';

import classnames from 'classnames';

import styles from './SearchBox.css';

export type SearchBoxProps = {
  className?: string;
};

export const SearchBox = memo(function SearchBox(props: SearchBoxProps): JSX.Element {
  const { className } = props;
  const searchBoxClassName = classnames(styles.searchBox, className);
  return (
    <div className={searchBoxClassName}>
      <input placeholder="What do you want to watch?" />
      <Button className={styles.button}>SEARCH</Button>
    </div>
  );
});
