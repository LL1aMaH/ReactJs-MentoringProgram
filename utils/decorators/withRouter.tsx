import React from 'react';

import { MemoryRouter } from 'react-router-dom';

export type TComponentFunc = () => JSX.Element;
export type TDecoratorLocation = {
  pathname?: string;
  search?: string;
};

const withRouter = (
  componentFunc: TComponentFunc,
  location: TDecoratorLocation = { pathname: '/' },
): JSX.Element => {
  return <MemoryRouter initialEntries={[location]}>{componentFunc()}</MemoryRouter>;
};

export default withRouter;
