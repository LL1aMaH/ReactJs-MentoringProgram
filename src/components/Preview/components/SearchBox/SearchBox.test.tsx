import React from 'react';

import { render, fireEvent } from '@testing-library/react';

import withRouter from '../../../../../utils/decorators/withRouter';

import { SearchBox } from './SearchBox';

describe('SearchBox', () => {
  it('renders correctly', () => {
    const component = render(withRouter(() => <SearchBox />));
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should correctly change input value', () => {
    const component = render(withRouter(() => <SearchBox />));

    const input = component.getByPlaceholderText('What do you want to watch?');
    const button = component.getByTestId('button');

    fireEvent.change(input, { target: { value: 'test' } });
    fireEvent.click(button);
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should correctly without change input value', () => {
    const component = render(withRouter(() => <SearchBox />));

    const input = component.getByPlaceholderText('What do you want to watch?');
    const button = component.getByTestId('button');

    fireEvent.change(input, { target: { value: '' } });
    fireEvent.click(button);

    expect(component.asFragment()).toMatchSnapshot();
  });
});
