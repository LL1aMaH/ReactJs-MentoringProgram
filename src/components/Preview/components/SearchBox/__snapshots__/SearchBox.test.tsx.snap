// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
<DocumentFragment>
  <div
    class="searchBox"
  >
    <input
      placeholder="What do you want to watch?"
      value=""
    />
    <button
      class="button button"
      data-testid="button"
      tabindex="0"
    >
      SEARCH
    </button>
  </div>
</DocumentFragment>
`;

exports[`SearchBox should correctly change input value 1`] = `
<DocumentFragment>
  <div
    class="searchBox"
  >
    <input
      placeholder="What do you want to watch?"
      value="test"
    />
    <button
      class="button button"
      data-testid="button"
      tabindex="0"
    >
      SEARCH
    </button>
  </div>
</DocumentFragment>
`;

exports[`SearchBox should correctly without change input value 1`] = `
<DocumentFragment>
  <div
    class="searchBox"
  >
    <input
      placeholder="What do you want to watch?"
      value=""
    />
    <button
      class="button button"
      data-testid="button"
      tabindex="0"
    >
      SEARCH
    </button>
  </div>
</DocumentFragment>
`;
